### ISSUE TITLE
Adding an Application section to Algorithm Table in README.md

### ISSUE THREAD

**Issue by aashutoshrathi** [iiitv/algos#148]
Should we have an application section for every implemented algorithm on this repository, which will help others (one who don't know about that particular algo) to figure out, how and where this algorithm will help them and will keep them interested.
For eg : 

## Implemented Algorithms

| Algorithm | C | CPP | Java | Python | Golang | Applications |
|:--------------|:----------------:|:----------------:|:----------------:|:-----------------:|:-----------------:|-----------------:|
| [Modular Exponential](http://www.geeksforgeeks.org/modular-exponentiation-power-in-modular-arithmetic/) | [\[X\]](https://github.com/iiitv/algos/blob/master/modular_exponential/modular_exponential.c) | | [\[X\]](https://github.com/iiitv/algos/blob/master/modular_exponential/ModularExponential.java) | [\[X\]](https://github.com/iiitv/algos/blob/master/modular_exponential/modular_exponential.py) | | [Efficient RSA-breaking Algorithm](https://crypto.stackexchange.com/questions/12513/solve-a-modular-exponentiation) |

**Comment by Anshuman-Verma** [comment]
*an application

**Comment by aashutoshrathi** [comment]
@Anshuman-Verma Thanks for such a valuable feedback 🙇

**Comment by Anshuman-Verma** [comment]
@aashutoshrathi i meant PR comment.

**Comment by aashutoshrathi** [comment]
@singhpratyush sir, take look at this

**Comment by singhpratyush** [comment]
I don't see any issue in this. Just that even the basic implementation would contain a solution to some problem.

The better thing to do would be to implement the methods in a file and then use them to solve the problems in other files.

@aviaryan: What do you think?

**Comment by aashutoshrathi** [comment]
So this issue should be close now ?
@singhpratyush

**Comment by singhpratyush** [comment]
No, we still need to have views of others.

**Comment by Monal5031** [comment]
Let's add a section like we added for DS, we'll implement a popular question related to that algorithm/DS there.?

**Comment by singhpratyush** [comment]
This is not that simple here. We would need to refactor the whole repo in order for this to fit in (because each algorithm/DS solves a problem, which would go in **Application** section if we follow this).

@aviaryan: What do you say about this?

**Comment by aviaryan** [comment]
Yes, we shouldn't have the "Applications" column with empty fields in `master`. If we are to do this, we will have to create a separate branch to do this, and this process will take some time. 
So we should create a new branch in this repo for adding the `Application` column. When complete and synced, we can merge it with master. 
@singhpratyush views?

**Comment by singhpratyush** [comment]
Also, almost all current implementations will fail which do now support different data types. Though this problem will be a nice exercise for everyone, it will become fairly difficult for newcomers to keep up.

**Comment by Monal5031** [comment]
@singhpratyush @aviaryan This can be closed now, or do we need more discussion over it?

**Comment by Anshuman-Verma** [comment]
Will it be worth to have a separate section for standard problems like, *rod cutting, coin change problem etc.* ?

**Comment by singhpratyush** [comment]
> almost all current implementations will fail which do now support different data types.

Please note that many implementations mentioned are not generic. So we need to change all the implementations to make them generic and make then implement the algorithm properly. Also, we will need to break down each implementation's directory into subdirectories based on language for a better organisation.

Also, there can be cases where different type of algorithms can use same base implementation to run. For example - DFS, BFS and A* searches can use same boilerplate code and just need to change the underlying data structure (queue, stack, priority queue respectively). How can this be achieved?

>  it will become fairly difficult for newcomers to keep up.

This repo aimed to help newcomers to get familiar with concepts of Git, open source, tests, etc. If we try to implement such things, the value of repository will surely increase as it will result in a better code. But will it retain its original spirit in that way?
